### ISSUE TITLE
What's the recommended way of using material-components-web-react components?

### ISSUE THREAD

**Issue by jimbojetlag** [material-components/material-components-web-react#61]
At this point it seems documentations for this library is nonexistent. The [general guidelines](https://github.com/material-components/material-components-web-react/blob/master/docs/guidelines.md) has a section about importing with ES5. Is that the recommended way of using this library?

It'd be nice to have an example of best practice in the documentations.

**Comment by jimbojetlag** [comment]
That guide is not specifically for React. When it comes to styles, most React libraries have an opinion about the styles approach. It is not clear to me which approach is preferred for this library.

**Comment by jimbojetlag** [comment]
Yes, that makes sense. But my point is something else. For example, create-react-app projects do not support importing sass files or some ES6 syntax by default, at least not until v2 is out.

There is a recent commit which transpiles ES6 to ES5 and sass to css, but it is now known to me how to customize the theme when importing css, as customizing by sass will be lost.

Ant Design React library ha s a detailed guide about how to use the library with create-react-app: https://ant.design/docs/react/use-with-create-react-app. I was hoping something similar could exist for this project to get a new user started without giving them even more decision fatigue in recent javascript world.

**Comment by moog16** [comment]
I was thinking that the #usage sections in each readme would be enough documentation for now: https://github.com/material-components/material-components-web-react/tree/master/packages/button#usage. 

Customizing is similar to MDC-Web. Our intent was to reuse the Sass mixins we've built in MDC-Web documented [here](https://github.com/material-components/material-components-web/blob/v0.35.0/packages/mdc-button/README.md#sass-mixins).

We've included the Sass within each package. However since we didn't compile the Sass into CSS, it might be easier to include the MDC-Web Sass/CSS files as necessary. So it looks like something like this:

```js
// Buttons need ripple and button styles
import '@material/ripple/mdc-ripple.scss';
import '@material/button/mdc-button.scss';

...

```

**Comment by jimbojetlag** [comment]
@moog16 are you suggesting that the when using **material-components-web-react**, sass imports should come directly from material-components-web? Will it be guaranteed that in the future the sass files in **material-components-web-react** will not have extra styles?

**Comment by moog16** [comment]
@jimbojetlag what do you mean extra styles? Do you mean duplicate?

**Comment by jimbojetlag** [comment]
@moog16 I don't mean duplicate, I mean any styles in addition to material-components-web styles that might have to be necessary for the React wrapper element to function.

**Comment by moog16** [comment]
@jimbojetlag I cannot guarantee anything as it this project is in alpha similar to MDC Web. But from my standpoint it doesn't make sense that MDC React will have differing styles. The goal of this project is to share as much code (Foundation JS & CSS) from MDC Web, effectively making MDC React a mirror of MDC Web's components. We currently follow this pattern and have no extra styles. The exception being Material Icon, but that is not an MDC Web component.

Short answer: you can expect MDC React to **NOT** have extra styles in addition to MDC Web's styles.
